examples/AspNetCore/Instrumentation.cs

```diff
@@ -0,0 +1,45 @@
+// <copyright file="Instrumentation.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace Examples.AspNetCore;
+
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+
+public class Instrumentation : IDisposable
+{
+    internal const string ScopeName = "Examples.AspNetCore";
```

Comment:
Exposing this since the `ActivitySource` and `Meter` registration is coupled to their names.

Reply:
I'd suggest to rename "ScopeName" to something else.
Scope is used in many context, so it could confuse new users.
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#get-a-tracer

Reply:
Do you have a suggestion? In my opinion this [instrumentation scope](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/glossary.md#instrumentation-scope) is the entity to which this data is associated. It also matches the [underlying proto](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/common/v1/common.proto#L71) (granted most users probably won't know that)

Reply:
okay to use ActivitySourceName, MeterName. 
AddSource, AddMeter usage is shown later, which looks easier to follow with AddMeter(instrumentaion.MeterNamer), AddActivitySource(instrument.ActivitySourceName)